src/Mvc/Mvc.ViewFeatures/src/ServerComponentSerializer.cs

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Text.Json;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures
+{
+    // **Component descriptor protocol**
```

Comment:
Is this all duplicated? I think it would be better to have one big comment, and then just mention where to find that from the other class.

Reply:
Made a "pointer" from here to "ServerComponentDeserializer.cs"